backend/config/routes.rb

```diff
@@ -151,10 +151,12 @@
       end
     end
 
-    resources :reports, only: [:index] do
+    resources :reports, only: :index do
       collection do
-        get :sales_total
-        post :sales_total
+        Rails.application.config.spree.reports.each do |report|
+          get report.to_s.demodulize.underscore.to_sym
+          post report.to_s.demodulize.underscore.to_sym
+        end
```

Comment:
Mentioned it also in the other comment, but an alternative here is to keep the existing code as it is, just add the Deprecated warning to both `get sales_total` and `post sales_total`, and add a `show` method action, which would work without meta-programming.